### PR TITLE
Support for new parameters in the Options and Route models

### PR DESCRIPTION
**Pull request by rogertsui0923**

### CODE REVIEW

**Comment by rogertsui0923** [comment]
The tests should pass now!

File: spec/route_spec.rb [hunk 1]
```diff
@@ -3,24 +3,10 @@
...
-
-  it "has number_of_unserved" do
-    expect(route.number_of_unserved).to eq(Factory::ROUTE_UNSERVED.count)
+  Factory::ROUTE_INPUT.each do |key, value|
+    it "has #{key}" do
+      expect(eval("route.#{key}")).to eq(value)
+    end
   end
 
   describe "#vehicleRoutes" do
```
Comment (oddg) [inline spec/route_spec.rb]: This test checks that calling `add_solution({})` sets an empty hash. That's a nice edge case to test but it would be nice to test the function with a real route too.

**Comment by rogertsui0923** [comment]
@oddg, thanks for the comment, I've added an additional test that checks that a solution hash is correctly converted into a hash of waypoint arrays.
